### ISSUE TITLE
Replacing Deprecation Trial with SAA/RWS for Multi-Product, Multi-Domain communication

### ISSUE THREAD

**Issue by Ang2User** [GoogleChromeLabs/privacy-sandbox-dev-support#244]
If a.com, b.com, and c.com are in the same RWS, and both a.com and b.com embed an **hidden iframe of c.com**, 
then can a.com and b.com use SAA to access **unpartitioned storage**  of those iframes of c.com.
(c.com provides authentication details such accessToken, refreshToken required by a.com and b.com)

**Comment by cfredric** [comment]
Yes, indirectly (and eventually, not necessarily today, depending on what kind of storage you want access to).

a.com and b.com cannot directly access storage that belongs to any other site (since doing so would violate the [Same Origin Policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy)). However, the c.com iframes can get access to their own unpartitioned storage and choose to share information with the parent frames via [postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) or similar.

To get access to c.com's unpartitioned storage within the iframes, a.com and b.com can call `document.requestStorageAccessFor('https://c.com')`, which grants `storage-access` permission to c.com under those sites. Then the hidden c.com iframes can call `document.requestStorageAccess()`, which will resolve (without requiring a user gesture) since permission was already granted by Chrome. Then the iframes can do what they wish with their unpartitioned data, including sharing it.

-----

Today, the Storage Access API in Chrome only supports getting access to unpartitioned cookies. However, we're working on extending this to all types of browser storage; please see https://developers.google.com/privacy-sandbox/blog/saa-non-cookie-storage and participate in the origin trial if you're interested.

**Comment by Ang2User** [comment]
Thanks for the update Fredric. It did clarify couple of uncertainties we had.

Just want to double check on the "Hidden c.om iframe" and its impact on SAA usage. I see in the documentation that 
"_The Storage Access API is one of the alternatives to using third-party cookies, so it is important to understand when to use this API compared to the others. It's meant for use cases where both the following are true:

**The user will interact with the embedded content—that is, it is not a passive iframe or a hidden iframe.**
The user has visited the embedded origin in a top-level context—that is, when that origin is not embedded in another site_."

Can you please confirm whether SAA will work our hidden iframe.

**Comment by cfredric** [comment]
> Can you please confirm whether SAA will work our hidden iframe.

If your use case is exclusively within the same RWS, and you are able to use `document.requestStorageAccessFor(origin)` in the top-level page first, then yes, the hidden iframes can call `document.requestStorageAccess()` and it will automatically resolve and grant them access (since `document.requestStorageAccessFor` already granted the `storage-access` permission).

If your sites are not in the same RWS and/or you cannot call `document.requestStorageAccessFor(origin)` first, then the Storage Access API isn't a viable option for you.
